Feature/delivery-address

San88Francisco opened the pull request.

San88Francisco committed:
created custom form and fetch api np

San88Francisco committed:
created api button 2

San88Francisco committed:
created autocomplete for city and added logic (getRequest, errors, loading and submit)

San88Francisco committed:
added/logic get city and created component Error

San88Francisco committed:
added logic autocomplete warehouse

San88Francisco committed:
added logic for autocoplete city

San88Francisco committed:
added logic disabled btn

pqoqubbw commented on src/components/Autocomplete/AutocompleteCustom.tsx:
```diff
@@ -0,0 +1,74 @@
+type PropsType = {
+  name: string
+  placeholder?: string
+  register: (name: string) => any
```
do not use `any`

San88Francisco replied:
done

pqoqubbw commented on src/components/Autocomplete/AutocompleteCustom.tsx:
```diff
@@ -0,0 +1,74 @@
+  placeholder?: string
+  register: (name: string) => any
+  options: string[]
+  onChange: (event: any, value: string) => void
```
same here

San88Francisco replied:
done

pqoqubbw commented on src/components/Autocomplete/AutocompleteCustom.tsx:
```diff
@@ -0,0 +1,74 @@
+  onChange: (event: any, value: string) => void
+  loading: boolean
+  setValueInput: (value: string) => void
+  errors: any
+  disabled?: any
```
same here

San88Francisco replied:
done

pqoqubbw commented on src/components/Autocomplete/AutocompleteCustom.tsx:
```diff
@@ -0,0 +1,74 @@
+      id="name"
+      sx={{ width: 300, mt: 2 }}
+      disableListWrap
+      onError={errors[name] ? () => {} : undefined}
```
`() => {}` what is it?

San88Francisco replied:
done

pqoqubbw commented on src/components/Autocomplete/AutocompleteCustom.tsx:
```diff
@@ -0,0 +1,74 @@
+      renderOption={(props, option, state) =>
+        [props, option, state.index] as ReactNode
+      }
+      renderGroup={(params) => params as any}
```
why do you need this?

San88Francisco replied:
This renderOption allows you to customize the display of the list 
enderOption={(props, option, state) =>
        [props, option, state.index] as ReactNode
      }
      But I deleted renderGroup, that param not required

pqoqubbw commented on src/components/Autocomplete/VariableSizeList.tsx:
```diff
@@ -0,0 +1,132 @@
+
+const LISTBOX_PADDING = 8
+
+function renderRow(props: ListChildComponentProps) {
+  const { data, index, style } = props
```
```suggestion
function renderRow({ data, index, style }: ListChildComponentProps) {
```

San88Francisco replied:
done

pqoqubbw commented on src/components/Autocomplete/VariableSizeList.tsx:
```diff
@@ -0,0 +1,132 @@
+  )
+}
+
+const OuterElementContext = createContext({})
```
forgot type for context

San88Francisco replied:
done

pqoqubbw commented on src/components/Autocomplete/VariableSizeList.tsx:
```diff
@@ -0,0 +1,132 @@
+  return <Box ref={ref} {...props} {...outerProps} />
+})
+
+function useResetCache(data: any) {
```
same issue with `any`

San88Francisco replied:
done

pqoqubbw commented on src/components/Autocomplete/VariableSizeList.tsx:
```diff
@@ -0,0 +1,132 @@
+export const ListboxComponent = forwardRef<
+  HTMLDivElement,
+  HTMLAttributes<HTMLElement>
+>(function ListboxComponent(props, ref) {
+  const { children, ...other } = props
```
```suggestion
>(function ListboxComponent({ children, ...other }, ref) {
```

San88Francisco replied:
done

pqoqubbw commented on src/components/Error/Error.tsx:
```diff
@@ -0,0 +1,22 @@
+import { FormHelperText } from '@mui/material'
+import { FC } from 'react'
+
+export const ErrorMessage: FC<{ error: any; styles?: any }> = ({
```
same issue with `any`

pqoqubbw commented on src/components/Modal-custom-btn/ModalCustomFormRadius.tsx:
```diff
@@ -0,0 +1,72 @@
+import * as Yup from 'yup'
+import RadioForm from '../RadioForm/RadioForm'
+
+interface ModalCustomFormRadiusProps {
```
you do not need `interface` here, use `type`

San88Francisco replied:
done

pqoqubbw commented on src/components/Modal-custom-btn/ModalCustomFormRadius.tsx:
```diff
@@ -0,0 +1,72 @@
+
+export const ModalCustomFormRadius: FC<ModalCustomFormRadiusProps> = ({
+  openModal = false,
+  setOpenModal = () => {}
```
do not mock functions like this

San88Francisco replied:
fine

pqoqubbw commented on src/components/Modal-custom-btn/ModalCustomFormRadius.tsx:
```diff
@@ -0,0 +1,72 @@
+    setError,
+    clearErrors,
+    formState: { errors }
+  } = useForm<any>({
```
same issue with `any`

pqoqubbw commented on src/components/Modal-custom-btn/ModalCustomFormRadius.tsx:
```diff
@@ -0,0 +1,72 @@
+    resolver: yupResolver(AddressRetention)
+  })
+
+  const onSubmit: SubmitHandler<any> = (data) => {
```
same issue with `any`

pqoqubbw commented on src/components/RadioForm/AddressNP.tsx:
```diff
@@ -0,0 +1,68 @@
+import { useNovaPoshtaCity } from '../../hooks/useNovaPoshtaCity'
+
+type PropsType = {
+  selectedValue: any
+  errors: any
```
same issue with `any`

pqoqubbw commented on src/components/RadioForm/AddressNP.tsx:
```diff
@@ -0,0 +1,68 @@
+  selectedValue: any
+  errors: any
+  setError: (name: string, error: { type: string; message: string }) => void
+  register: (name: string) => any
```
same issue with `any`

San88Francisco replied:
done

pqoqubbw commented on src/components/RadioForm/AddressNP.tsx:
```diff
@@ -0,0 +1,68 @@
+  )
+  const [street, setStreet] = useState<string>('Соборна')
+
+  const onChangeCity = (_: any, value: string) => {
```
same issue with `any`

San88Francisco replied:
done

pqoqubbw commented on src/components/RadioForm/AddressNP.tsx:
```diff
@@ -0,0 +1,68 @@
+    }
+  }
+
+  const onChangeStreet = (_: any, value: string) => {
```
same issue with `any`

pqoqubbw commented on src/components/RadioForm/RadioForm.tsx:
```diff
@@ -0,0 +1,62 @@
+
+type PropsType = {
+  children: ReactNode
+  register: any
+  setValue: any
+  errors: any
+  setError: any
+  clearErrors: any
```
same issue with `any`

San88Francisco replied:
done

pqoqubbw commented on src/components/RadioForm/SeparationNP.tsx:
```diff
@@ -0,0 +1,176 @@
+import axiosInstance from '../../axios/settings'
+
+type PropsType = {
+  selectedValue: any
+  errors: any
+  register: (name: string) => any
```
same issue with `any`

San88Francisco replied:
done

pqoqubbw commented on src/components/RadioForm/SeparationNP.tsx:
```diff
@@ -0,0 +1,176 @@
+  clearErrors: (name: string) => void
+}
+
+const getCityRef = (value: string, novaPoshtaCity: any[]) => {
```
same issue with `any`

San88Francisco replied:
done

pqoqubbw commented on src/components/RadioForm/SeparationNP.tsx:
```diff
@@ -0,0 +1,176 @@
+  return selectedCity ? selectedCity.Ref : null
+}
+
+const getDefaultCityRef = (value: string, cityDefault: any[]) => {
```
same issue with `any`

San88Francisco replied:
done

pqoqubbw commented on src/components/RadioForm/SeparationNP.tsx:
```diff
@@ -0,0 +1,176 @@
+  const MAX_LENGTH = 50
+
+  const optionsDataCity = useMemo(() => {
+    return warehouses.map(
+      (warehouse) =>
+        warehouse.address_warehouse
+          .replace(/\(до 30 кг на одне місце\)/g, '')
+          .replace(/\(до 30 кг\)/g, '')
+          .replace(/\(до 10 кг\)/g, '')
+          .replace(/\(до 5 кг\)/g, '')
+          .replace(/\(до 200 кг\)/g, '')
+          .replace(/\(до 1100 кг \)/g, '')
+          .replace(/\n/g, '')
+          .replace(/№(\d+)\s*:/g, '№$1:')
+          .trim()
+          .slice(0, MAX_LENGTH) +
+        (warehouse.address_warehouse.length > MAX_LENGTH ? '...' : '')
+    )
```
move this to `utils` as a helper functions

San88Francisco replied:
done

pqoqubbw commented on src/components/RadioForm/SeparationNP.tsx:
```diff
@@ -0,0 +1,176 @@
+  }, [warehouses])
+
+  const onChangeCity = useCallback(
+    (_event: any, value: string) => {
```
same issue with `any`

San88Francisco replied:
done

pqoqubbw commented on src/components/RadioForm/SeparationNP.tsx:
```diff
@@ -0,0 +1,176 @@
+  )
+
+  const onChangeWarehouse = useCallback(
+    (_event: any, value: string) => {
```
same issue with `any`

San88Francisco replied:
done

pqoqubbw commented on src/hooks/useNovaPoshtaCity.ts:
```diff
@@ -0,0 +1,110 @@
+import { useState, useEffect, useMemo } from 'react'
+
+const cityDefault = [
```
```suggestion
const CITY_DEFAULT = [
```

San88Francisco replied:
done

pqoqubbw commented on src/hooks/useNovaPoshtaCity.ts:
```diff
@@ -0,0 +1,110 @@
+  { name: 'Рівне', ref: 'e71d65e1-4b33-11e4-ab6d-005056801329' }
+]
+const fetchNovaPoshtaCity = async (cityName: string) => {
+  const apiKey = 'f07607422838cfac21a0d1b8603086ca'
```
move this to `env`

San88Francisco replied:
done

pqoqubbw commented on src/hooks/useNovaPoshtaCity.ts:
```diff
@@ -0,0 +1,110 @@
+}
+
+export const useNovaPoshtaCity = () => {
+  const [novaPoshtaCity, setNovaPoshtaCity] = useState<any[]>([])
```
same issue with `any`

San88Francisco replied:
done

pqoqubbw commented on src/hooks/useNovaPoshtaCity.ts:
```diff
@@ -0,0 +1,110 @@
+    }
+  }
+
+  const cityRenderNP = useMemo(
+    () =>
+      novaPoshtaCity.length > 0
+        ? novaPoshtaCity[0].Addresses.map(
+            (address: any) =>
+              `${address.SettlementTypeCode} ${address.MainDescription} (${address.Area} обл.)`
+          )
+        : [],
+    [novaPoshtaCity]
+  )
+  const options = useMemo(() => {
+    if (!valInputCity) return cityDefault.map((city) => city.name)
+    const filteredCities = cityDefault
+      .map((city) => city.name)
+      .filter((city) => city.toLowerCase().includes(valInputCity.toLowerCase()))
+    return filteredCities.length > 0 ? filteredCities : cityRenderNP
+  }, [valInputCity, cityRenderNP])
```
try to simplify these functions cause now I can not understand what you do

San88Francisco replied:
I optimization code and deleted this fun cityRenderNP

pqoqubbw commented on src/hooks/useNovaPoshtaCity.ts:
```diff
@@ -0,0 +1,110 @@
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      if (
+        valInputCity &&
+        valInputCity.length > 3 &&
+        !cityDefault.some((city) => city.name === valInputCity) &&
+        !/[()]/.test(valInputCity)
```
same issue with complexity here

San88Francisco replied:
removed

pqoqubbw commented on src/hooks/useNovaPoshtaCity.ts:
```diff
@@ -0,0 +1,110 @@
+        setNovaPoshtaCity([])
+        handleCityFetch(valInputCity)
+      }
+    }, 1000)
```
move this number as `const`

San88Francisco committed:
Added logic for warehouse

San88Francisco committed:
added useNovaPoshtaLocations

San88Francisco committed:
opimization code

San88Francisco committed:
get id in useFrom

San88Francisco committed:
created component postomats and added logic api

San88Francisco committed:
corected code

San88Francisco committed:
Merge branch 'master' of https://github.com/sushka2023/sushka-shop-fe into feature/form-address

San88Francisco committed:
deBug merge

San88Francisco committed:
added form address

San88Francisco committed:
fix/ correction code

San88Francisco committed:
feature/ visualization card address
- created AddressDetails
- added sending data from the form (department, post office and address)
- added output of address cards

San88Francisco committed:
feature/ delete card address

San88Francisco committed:
fix/ refactored the code

San88Francisco committed:
Merge branch 'master' of https://github.com/sushka2023/sushka-shop-fe into feature/form-address

San88Francisco committed:
added/ env file api_key

pqoqubbw commented on .env:
do not push `.env` file to git

San88Francisco replied:
done

pqoqubbw commented on src/components/Account-panel/Delivery-address/AddressDetails.tsx:
```diff
@@ -0,0 +1,69 @@
+  cleanedAddress: string
+}
+
+const getLogoComponent = (isUkrPoshta: boolean) =>
+  isUkrPoshta ? LogoUrk : LogoNP
+
+const getNovaPoshtaType = (addressWarehouse: string) => {
+  if (addressWarehouse.includes('Поштомат')) return 'Поштомат'
+  if (addressWarehouse.includes('Відділення')) return 'Відділення'
+  return 'Адресна'
+}
+
+const getCityArea = (city: string, area?: string) =>
+  `${city}${area ? `, ${area}` : ''}`
+
+const getStreetAddress = (street?: string, houseNumber?: string) =>
+  `${street ?? ''} ${houseNumber ?? ''}`
+
+const getApartmentFloor = (apartmentNumber?: string, floor?: number) =>
+  `${apartmentNumber ? `, кв. ${apartmentNumber}` : ''}${floor ? `, поверх ${floor}` : ''}`
+
+const getFormattedAddress = (
+  address: AddressDetailsType,
+  cleanedAddress: string
+) => {
+  const cityArea = getCityArea(address.city, address.area)
+  const streetAddress = getStreetAddress(address.street, address.house_number)
+  const apartmentFloor = getApartmentFloor(
+    address.apartment_number,
+    address.floor
+  )
+
+  return `${cityArea}${cleanedAddress ? `, ${cleanedAddress}` : `, ${streetAddress}${apartmentFloor}`}`
+}
+
```
move all of this to local helper file

San88Francisco replied:
done

pqoqubbw commented on src/components/RadioForm/BranchesNP.tsx:
```diff
@@ -0,0 +1,164 @@
+      }
+    },
+    [
+      setValue,
+      setValInputCity,
+      novaPoshtaCity,
+      CITY_DEFAULT,
+      clearErrors,
+      getDefaultCityRef,
+      getCityRef,
+      setSettleRef
```
remove all setters

San88Francisco replied:
done

pqoqubbw commented on src/components/RadioForm/PostomatsNP.tsx:
```diff
@@ -0,0 +1,153 @@
+      setNewRequest(true)
+      clearErrors('postomats')
+    }
+  }, [isDisabled, setValue, setNewRequest, clearErrors])
```
same here

San88Francisco replied:
done

San88Francisco committed:
fix/ took the delete icon on the component with MUI

San88Francisco committed:
fix/ import

pqoqubbw commented on .env:
do not push this file to github, this should be local only

San88Francisco replied:
done

San88Francisco committed:
fix

San88Francisco committed:
fix
